File: snippets/react-native/parsing_inputs.ts
import {
  BindingLiquidSdk,
  InputType_Tags
} from '@breeztech/breez-sdk-liquid-react-native'

const parseInputs = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: parse-inputs
  const input = 'an input to be parsed...'

  const parsed = sdk.parse(input)

  switch (parsed.tag) {
    case InputType_Tags.BitcoinAddress:
      console.log(`Input is Bitcoin address ${parsed.inner.address.address}`)
      break

    case InputType_Tags.Bolt11:
      console.log(
        `Input is BOLT11 invoice for ${parsed.inner.invoice.amountMsat != null
          ? JSON.stringify(parsed.inner.invoice.amountMsat)
          : 'unknown'
        } msats`
      )
      break

    case InputType_Tags.Bolt12Offer:
      console.log(
        `Input is BOLT12 offer for min ${JSON.stringify(parsed.inner.offer.minAmount)} msats - BIP353 was used: ${parsed.inner.bip353Address != null}`
      )
      break

    case InputType_Tags.LnUrlPay:
      console.log(
        `Input is LNURL-Pay/Lightning address accepting min/max ${parsed.inner.data.minSendable}/${parsed.inner.data.maxSendable} msats - BIP353 was used: ${parsed.inner.bip353Address != null}`
      )
      break

    case InputType_Tags.LnUrlWithdraw:
      console.log(
        `Input is LNURL-Withdraw for min/max ${parsed.inner.data.minWithdrawable}/${parsed.inner.data.maxWithdrawable} msats`
      )
      break

    default:
      // Other input types are available
      break
  }
  // ANCHOR_END: parse-inputs
}

File: snippets/swift/BreezSDKExamples/Sources/ParsingInputs.swift
import BreezSDKLiquid
import Foundation

func parseInput(sdk: BindingLiquidSdk) {
    // ANCHOR: parse-inputs
    let input = "an input to be parsed..."

    do {
        let inputType = try sdk.parse(input: input)
        switch inputType {
        case .bitcoinAddress(let address):
            print("Input is Bitcoin address \(address.address)")

        case .bolt11(let invoice):
            let amount = invoice.amountMsat.map { String($0) } ?? "unknown"
            print("Input is BOLT11 invoice for \(amount) msats")

        case .bolt12Offer(let offer, let bip353Address):
            print("Input is BOLT12 offer for min \(offer.minAmount) msats - BIP353 was used: \(bip353Address != nil)")

        case .lnUrlPay(let data, let bip353Address):
            print(
                "Input is LNURL-Pay/Lightning address accepting min/max \(data.minSendable)/\(data.maxSendable) msats - BIP353 was used: \(bip353Address != nil)"
            )
        case .lnUrlWithdraw(let data):
            print(
                "Input is LNURL-Withdraw for min/max \(data.minWithdrawable)/\(data.maxWithdrawable) msats"
            )

        default:
            break  // Other input types are available
        }
    } catch {
        print("Failed to parse input: \(error)")
    }
    // ANCHOR_END: parse-inputs
}

